### PR TITLE
Update kubeconfig topics.

### PR DESCRIPTION
**Pull request by steveperry-53**
The purpose of this PR is to:
* Refactor the topics on kubeconfig files. Now there is one Task topic and one Concept topic.
* In the new Task topic, provide steps that the reader can do as they read the  topic.
* Address https://github.com/kubernetes/kubernetes.github.io/issues/4496.
* Address https://github.com/kubernetes/kubernetes/issues/17658.

New topics:
* [Configure Access to Multiple Clusters]( https://deploy-preview-4439--kubernetes-io-user-journeys.netlify.com/docs/tasks/access-application-cluster/configure-access-multiple-clusters/)
* [Organize Cluster Access Using kubeconfig Files]( https://deploy-preview-4439--kubernetes-io-user-journeys.netlify.com/docs/concepts/configuration/organize-cluster-access-kubeconfig/)

Deleted topics:
* [Authenticate Across Clusters with kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/authenticate-across-clusters-kubeconfig/)
* [Share Cluster Access with kubeconfig](https://kubernetes.io/docs/tasks/administer-cluster/share-configuration/)

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4439)

### CODE REVIEW

File: docs/concepts/configuration/merge-kubeconfig-files.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+   1. Use a command-line flag if it exists: `--user` or `--cluster`.
+   1. If the context is non-empty, take the user or cluster from the context.
+
+   The user and cluster can be empty at this point.
+
+1. Determine the actual cluster information to use. At this point, there might or
+   might not be cluster information. 
+   Build each piece of the cluster information based on this chain; the first hit wins:
+
+   1. Use command line flags if they exist: `--server`, `--api-version`, `--certificate-authority`, `--insecure-skip-tls-verify`.
```
Comment (tengqm) [inline docs/concepts/configuration/merge-kubeconfig-files.md]: the command line option `--api-version` is gone ...

Comment (steveperry-53) [inline docs/concepts/configuration/merge-kubeconfig-files.md]: Removed --api-version.

**Comment by steveperry-53** [comment]
@tengqm, Thanks for the review. I removed the --api-version flag. Does this PR look OK for merging now?

File: docs/concepts/configuration/organize-cluster-access-kubeconfig.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+cluster and namespace listed in the current context. And it uses the
+credentials of the user listed in the current context.
+
+## The KUBECONFIG environment variable
+
+The `KUBECONGIG` environment variable holds a list of kubeconfig files.
+For Linux and Mac, the list is colon-delimited. For Windows, the list
+is semicolon-delimited. The `KUBECONFIG` environment variable is not
+required. If the `KUBECONFIG` environment variable doesn't exist,
+`kubectl` uses the default kubeconfig file, `HOME/.kube/config`.
```
Comment (steveperry-53) [inline docs/concepts/configuration/organize-cluster-access-kubeconfig.md]: Fixed.

**Pull request merged**